input/system/selfhosted/syslog_log_receiver.go

```diff
@@ -0,0 +1,64 @@
+package selfhosted
+
+import (
+	"context"
+	"regexp"
+	"strconv"
+	"time"
+	
+	"gopkg.in/mcuadros/go-syslog.v2"
+
+	"github.com/pganalyze/collector/util"
+)
+
+func setupSyslogHandler(ctx context.Context, logSyslogServer string, out chan<- SelfHostedLogStreamItem, prefixedLogger *util.Logger) error {
+	channel := make(syslog.LogPartsChannel)
+	handler := syslog.NewChannelHandler(channel)
+
+	server := syslog.NewServer()
+	server.SetFormat(syslog.RFC5424)
+	server.SetHandler(handler)
+	err := server.ListenTCP(logSyslogServer)
+	if err != nil {
+		return err
+	}
+	server.Boot()
+
+	go func(ctx context.Context, server *syslog.Server, channel syslog.LogPartsChannel) {
+		for {
+			select {
+			case logParts := <-channel:
+				item := SelfHostedLogStreamItem{}
+				item.Line, _ = logParts["message"].(string)
+
+				item.OccurredAt, _ = logParts["timestamp"].(time.Time)
+
+				pidStr, _ := logParts["proc_id"].(string)
+				if s, err := strconv.ParseInt(pidStr, 10, 32); err == nil {
+					item.BackendPid = int32(s)
+				}
+
+				logLineNumberStr, _ := logParts["structured_data"].(string)
+				logLineNumberParts := regexp.MustCompile(`^\[(\d+)-(\d+)\]$`).FindStringSubmatch(logLineNumberStr)
```

Comment:
Should we lift this regexp initialization? In terms of performance it probably doesn't matter (and/or Go is smart enough to optimize that itself), but it would be a little more readable.

Reply:
Good idea - I'll pull this out for better readability.